frontend/src/components/teacher/student-management/classrooms/ClassroomCard.tsx

```diff
@@ -72,18 +83,22 @@ const ClassroomCard = ({
         w="100%"
       >
         <HStack alignItems="flex-start" justifyContent="space-between" w="100%">
-          <LinkOverlay
-            as={RouterLink}
-            to={{
-              pathname: Routes.DISPLAY_CLASSROOM_PAGE(id),
-              state: { className: name, startDate, grade },
-            }}
-          >
-            <Text color="grey.400" textStyle="mobileHeader3">
-              {name}
-            </Text>
-          </LinkOverlay>
-          <ClassroomPopover />
+          {disabledOptions ? (
+            classroomTitle
+          ) : (
+            <>
+              <LinkOverlay
+                as={RouterLink}
+                to={{
+                  pathname: Routes.DISPLAY_CLASSROOM_PAGE(id),
+                  state: { className: name, startDate, grade },
+                }}
+              >
+                {classroomTitle}
+              </LinkOverlay>
+              <ClassroomPopover />
+            </>
+          )}
```

Comment:
Same here, I didn't think the differences warranted a separate component, so just tried to make this one accommodate the minor differences from this view.